travel-planner/travel-planner-ui/src/Components/Maps/Maps.jsx

```diff
@@ -37,44 +33,91 @@ export default function Maps({
 
 		// get the markers
 		useEffect(async () => {
-			const res = await axios.post(`http://localhost:${PORT}/maps/getMarkers`, {
-				mapId: trip,
-			});
-			setMarkers(res.data);
-		}, []);
-
-		if (!markers) {
-			return <Loading />;
-		}
+			try {
+				const res = await axios.post(
+					`http://localhost:${PORT}/maps/getMarkers`,
+					{
+						mapId: trip.objectId,
+					}
+				);
+				setMarkers(res.data.markers);
+				setUpdate(false);
+			} catch (error) {
+				console.log("Error fetching markers: ", error);
+			}
+		}, [update]);
 
 		return (
 			<div
 				id="displayed-map"
 				ref={ref}
 				style={{ flexGrow: "1", height: "100%" }}
 			>
-				{markers.map((marker) => (
-					<Marker
-						position={{
-							lat: marker.Location.latitude,
-							lng: marker.Location.longitude,
-						}}
-						title={marker.Name}
-						icon={image}
-						map={map}
-						key={marker.objectId}
-					/>
-				))}
+				{markers &&
+					markers.map((marker) => (
+						<Marker
+							position={{
+								lat: marker.Location.latitude,
+								lng: marker.Location.longitude,
+							}}
+							title={marker.Name}
+							icon={image}
+							map={map}
+							key={marker.objectId}
+							objectId={marker.objectId}
+							address={marker.Address}
+							content={marker.Content}
+							PORT={PORT}
+							setCurNote={setCurNote}
+						/>
+					))}
 			</div>
 		);
 	};
 
+	const addNewMarker = async () => {
+		// we really will need to get error checking
+
+		// 1. reformat address so it can be passed to the api
+		// Don't think this api needs it
+		// 2. user axios post request address: address
+		// Can we combine this into one post request please?
+		const addressRes = await axios.post(
+			`http://localhost:${PORT}/maps/getAddress`,
+			{
+				address: address,
+			}
+		);
+		const location = addressRes.data.coordinates; // lets reformat how I recieve this information
+		const reformattedAddress = addressRes.data.address; // these are both undefined
+
+		// 3. call addMarker
+		const newMarkerRes = await axios.post(
```

Comment:
I don't follow. Is this needed? Or is it needed afterwards? Looks like there are 5 steps in this function, are they all implemented? If not, can we mark the to-do steps with "todo" for better clarity?

Reply:
These were comments from before I missed after fixing the backend /getAddress endpoint. These now work and do not return undefined. I've removed the comment for the next pr.

Reply:
All the steps for this have been implemented. I left the steps here in the case that we need to switch the api later on.

Reply:
Ah gotcha. Do we want to save newMarkerRes? It's not used @KimberlyTruong

Reply:
Nice catch! We don't. I've removed it.